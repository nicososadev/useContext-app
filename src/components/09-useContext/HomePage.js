import React, { useContext } from 'react'
import { userContext } from './UserContext'

export const HomePage = () => {

    const { user } = useContext(userContext)

    return (
        <div>
            <h1>Homepage</h1>
            <hr />

            <pre>
                { JSON.stringify(user, null, 3) }
            </pre>
            
        </div>
    )
}
