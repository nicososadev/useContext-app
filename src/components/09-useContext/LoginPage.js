import React, { useContext } from 'react'
import { userContext } from './UserContext'

export const LoginPage = () => {

    const { setUser } = useContext(userContext)

    return (

        <div>
            <h1>Login</h1>
            <hr />
            <button 
                className="btn btn-primary"
                onClick={ () => setUser({
                    id: 1,
                    name: 'Nicolas'
                })}
            >
                Login
            </button>
        </div>
    )
}
