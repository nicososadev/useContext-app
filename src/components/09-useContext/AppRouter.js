import React from 'react'
import { BrowserRouter as Router, Switch, Route } from "react-router-dom"
import { AboutPage } from './AboutPage'
import { HomePage } from './HomePage'
import { LoginPage } from './LoginPage'
import { NavBar } from './NavBar'

export const AppRouter = () => {

    return (
        <Router>
            <div>

                <NavBar />
                <div className="container">
                    <Switch>
                        <Route exact  path="/" component={ HomePage }/>

                        <Route exact path="/login" component={ LoginPage }/>
                        <Route exact path="/about" component={ AboutPage }/>

                        <Route component={ HomePage } />
                    </Switch>
                </div>
            </div>
        </Router>
    )
}
