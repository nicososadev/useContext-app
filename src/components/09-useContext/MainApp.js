import React, { useState } from 'react'
import { AppRouter } from './AppRouter'
import { userContext } from './UserContext'

export const MainApp = () => {
    
    const [user, setUser] = useState({})

    return (
        <userContext.Provider value={{
            user,
            setUser
        }}>

            <AppRouter />

        </userContext.Provider>
    )
}
