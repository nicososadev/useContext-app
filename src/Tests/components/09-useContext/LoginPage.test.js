import React from 'react'
import '@testing-library/jest-dom'
import { mount } from 'enzyme'
import { LoginPage } from '../../../components/09-useContext/LoginPage'
import { userContext } from '../../../components/09-useContext/UserContext'

describe('Pruebas en el componente LoginPage', () => {

    const setUser = jest.fn()

    const wrapper = mount( 
        <userContext.Provider value={{ setUser }}>

            <LoginPage />)

        </userContext.Provider>
    )
    
    test('Debe mostrarse correctamente', () => {
        
        expect( wrapper ).toMatchSnapshot()
    })
    
    test('Debe llamar a la funcion setUser con su respectivo argumento', () => {
        
        const user = {
            id: 1,
            name: 'Nicolas'
        }

        wrapper.find('button').simulate('click')

        expect( setUser ).toHaveBeenCalledWith( user )

    })
    
})
