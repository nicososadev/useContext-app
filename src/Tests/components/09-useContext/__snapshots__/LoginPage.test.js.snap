// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente LoginPage Debe mostrarse correctamente 1`] = `
Array [
  <div>
    <h1>
      Login
    </h1>
    <hr />
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      Login
    </button>
  </div>,
  ")",
]
`;
